src/Microsoft.ML.ImageAnalytics/ImagePixelExtractorTransform.cs

```diff
@@ -476,26 +478,26 @@ private ValueGetter<VBuffer<TValue>> GetGetterCore<TValue>(IRow input, int iinfo
 
                         if (src == null)
                         {
-                            dst = new VBuffer<TValue>(size, 0, dst.Values, dst.Indices);
+                            dst = VBufferMutationContext.Create(ref dst, size, 0)
+                                .CreateBuffer();
                             return;
                         }
 
                         Host.Check(src.PixelFormat == System.Drawing.Imaging.PixelFormat.Format32bppArgb);
                         Host.Check(src.Height == height && src.Width == width);
 
-                        var values = dst.Values;
-                        if (Utils.Size(values) < size)
-                            values = new TValue[size];
+                        var mutation = VBufferMutationContext.Create(ref dst, size);
+                        var values = mutation.Values;
 
                         float offset = ex.Offset;
                         float scale = ex.Scale;
                         Contracts.Assert(scale != 0);
 
-                        var vf = values as float[];
-                        var vb = values as byte[];
-                        Contracts.Assert(vf != null || vb != null);
+                        Span<float> vf = typeof(TValue) == typeof(float) ? MemoryMarshal.Cast<TValue, float>(mutation.Values) : default;
```

Comment:
>vf [](start = 36, length = 2)

I feel that a more common pattern is to have 2 getters: one for float case and one for byte case. 
Maybe add a `// REVIEW: split the getter into 2 specialized getters, one for float case and one for byte case` ?

Reply:
Done.

---
In reply to: [233673911](https://github.com/dotnet/machinelearning/pull/1580#discussion_r233673911) [](ancestors = 233673911)